### PR TITLE
Implement stepDefinitions for login feature

### PR DESCRIPTION
**Pull request by niteshrajbaral**
This PR implements stepDefinitions with page object model for login feature test.

### CODE REVIEW

File: tests/stepDefinitions/loginContext.js [hunk 1]
```diff
@@ -1,35 +1,39 @@
...
+  await page.goto("http://127.0.0.1:3000/admin/login")
+  await expect(page.url()).toBe("http://127.0.0.1:3000/admin/login");
 });
 
-When('user {string} logs in with following credentials', function (string, dataTable) {
-  console.log(" When ");
+When('user {string} logs in with following credentials',async function (string, dataTable) {
   for (const { email, password } of dataTable.hashes()) {
-    console.log(email, password)
+    await page.locator(emailFieldLocator).fill(email);
```
Comment (prashant-gurung899) [inline tests/stepDefinitions/loginContext.js:16]: Just a suggestion!!!
You can use this alternative as well
```suggestion
    await page.fill(emailFieldLocator, email);
```

File: tests/stepDefinitions/loginContext.js [hunk 2]
```diff
@@ -1,35 +1,39 @@
...
+    await page.locator(passwordFieldLocator).fill(password);
+    await page.locator(loginBtnLocator).click();
```
Comment (prashant-gurung899) [inline tests/stepDefinitions/loginContext.js:18]:
```suggestion
    await page.click(loginBtnSelector);
```

File: tests/stepDefinitions/loginContext.js [hunk 3]
```diff
@@ -1,35 +1,39 @@
 const { Given, When, Then } = require('@cucumber/cucumber')
 const { expect } = require("@playwright/test");
-
-
-// for valid scenario
-Given('user {string} has navigated to the admin login page', function (string) {
-  console.log(" Given ");
+const loginBtnLocator=".btn.btn-block.btn-primary"
```
Comment (prashant-gurung899) [inline tests/stepDefinitions/loginContext.js:3]: Always give one space before and after '='
```suggestion
const loginBtnLocator = ".btn.btn-block.btn-primary"
```

File: tests/stepDefinitions/loginContext.js [hunk 4]
```diff
@@ -1,35 +1,39 @@
...
+const invalidErrorSelector=".alert-danger"
+
+Given('user {string} has navigated to the admin login page', async function (string) {
+  await page.goto("http://127.0.0.1:3000/admin/login")
+  await expect(page.url()).toBe("http://127.0.0.1:3000/admin/login");
 });
 
-When('user {string} logs in with following credentials', function (string, dataTable) {
-  console.log(" When ");
+When('user {string} logs in with following credentials',async function (string, dataTable) {
```
Comment (prashant-gurung899) [inline tests/stepDefinitions/loginContext.js:14]:
```suggestion
When('user {string} logs in with following credentials',async function (string, credentials) {
```

File: tests/stepDefinitions/loginContext.js [hunk 5]
```diff
@@ -1,35 +1,39 @@
 const { Given, When, Then } = require('@cucumber/cucumber')
 const { expect } = require("@playwright/test");
```
Comment (anon-pradip) [inline tests/stepDefinitions/loginContext.js:2]:
```suggestion
const { Given, When, Then } = require('@cucumber/cucumber')
const { expect } = require("@playwright/test");

```

File: tests/stepDefinitions/loginContext.js [hunk 6]
```diff
@@ -1,35 +1,39 @@
...
-
-
-// for valid scenario
-Given('user {string} has navigated to the admin login page', function (string) {
-  console.log(" Given ");
+const loginBtnLocator=".btn.btn-block.btn-primary"
+const emailFieldLocator="#spree_user_email"
+const passwordFieldLocator="#spree_user_password"
+const dashboardSelector="//*[@id='missingHomepage']/div/div[2]/h4[1]"
```
Comment (anon-pradip) [inline tests/stepDefinitions/loginContext.js:6]: If you want to go for that `h1`, then you can simply use: `//*[@id='missingHomepage']//div//h4[1]`
or
we can use this &darr;
const dashboardSelector = `//div[@id='missingHomepage']//h1`

File: tests/pageObjects/LoginPage.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
+class LoginPage{
+    constructor(){
+        this.emailFieldSelector = "#spree_user_email"
+        this.loginBtnSelector = ".btn.btn-block.btn-primary"
+        this.passwordFieldSelector = "#spree_user_password"
+        this.dashboardSelector = "//div[@id='missingHomepage']//h1"
+        this.invalidErrorSelector = ".alert-danger"
```
Comment (prashant-gurung899) [inline tests/pageObjects/LoginPage.js:7]:
```suggestion
        this.errorMessageSelector = ".alert-danger"
```

File: tests/pageObjects/LoginPage.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+    }
```
Comment (prashant-gurung899) [inline tests/pageObjects/LoginPage.js:8]:
```suggestion
    }
    
```

File: tests/pageObjects/LoginPage.js [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+    async navigate() {
+        await page.goto("http://127.0.0.1:3000/admin/login");
+    }
```
Comment (prashant-gurung899) [inline tests/pageObjects/LoginPage.js:11]:
```suggestion
    }
    
```

File: tests/pageObjects/LoginPage.js [hunk 10]
```diff
@@ -0,0 +1,20 @@
+class LoginPage{
+    constructor(){
+        this.emailFieldSelector = "#spree_user_email"
+        this.loginBtnSelector = ".btn.btn-block.btn-primary"
+        this.passwordFieldSelector = "#spree_user_password"
+        this.dashboardSelector = "//div[@id='missingHomepage']//h1"
+        this.invalidErrorSelector = ".alert-danger"
+    }
+    async navigate() {
```
Comment (prashant-gurung899) [inline tests/pageObjects/LoginPage.js:9]:
```suggestion
    async navigateToAdminLoginPage() {
```

**Comment by prashant-gurung899** [comment]
Please update your description.

File: tests/pageObjects/LoginPage.js [hunk 11]
```diff
@@ -0,0 +1,23 @@
+class LoginPage{
+    constructor(){
+        this.emailFieldSelector = "#spree_user_email"
+        this.loginBtnSelector = ".btn.btn-block.btn-primary"
+        this.passwordFieldSelector = "#spree_user_password"
+        this.dashboardSelector = "//div[@id='missingHomepage']//h1"
+        this.errorErrorSelector = ".alert-danger"
```
Comment (prashant-gurung899) [inline tests/pageObjects/LoginPage.js:7]:
```suggestion
        this.errorMessageSelector = ".alert-danger"
```

File: tests/pageObjects/LoginPage.js [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+    }
+
+    async navigateToAdminLoginPage() {
+        await page.goto("http://127.0.0.1:3000/admin/login");
+    }
+
+    async login(email, password) {
+        await page.type(this.emailFieldSelector, email);
```
Comment (prashant-gurung899) [inline tests/pageObjects/LoginPage.js:15]: Use `fille`
```suggestion
        await page.fill(this.emailFieldSelector, email);
```

**Pull request merged**